### PR TITLE
Drop constraints before dropping tables

### PR DESCRIPTION
**Pull request by jayvdb**
Allows https://github.com/Electron100/butane/pull/252 to pass

### CODE REVIEW

File: butane_core/src/db/pg.rs [hunk 1]
```diff
@@ -583,6 +596,16 @@ fn create_table_constraints(table: &ATable) -> String {
         .join("\n")
 }
 
+fn remove_table_constraints(table: &ATable) -> Result<String> {
```
Comment (Electron100) [inline butane_core/src/db/pg.rs:599]: Method name is a bit misleading. It doesn't remove all constraints, it removes only the foreign key constraints. Not unique/primary key/etc.

File: butane_core/src/db/pg.rs [hunk 2]
```diff
@@ -551,9 +551,22 @@ fn sql_for_op(current: &mut ADB, op: &Operation) -> Result<String> {
         Operation::AddTableConstraints(table) => Ok(create_table_constraints(table)),
         Operation::AddTableIfNotExists(table) => Ok(create_table(table, true)?),
         Operation::RemoveTable(name) => Ok(drop_table(name)),
+        Operation::RemoveTableConstraints(table) => remove_table_constraints(table),
```
Comment (Electron100) [inline butane_core/src/db/pg.rs:554]: Why does this need to be a separate operation semantically? Why not just do it automatically as part of `RemoveTable` handling? It looks like ADB `diff` is always pairing them anyway, and I would argue that whether a table's constraints have to be dropped before the table can be dropped is a backend-specific attribute. Theoretically there could be a backend for which that wasn't true.

Comment (Electron100) [reply butane_core/src/db/pg.rs:554]: Would have to change the signature of `RemoveTable` slightly, but I think that's OK

Comment (jayvdb) [reply butane_core/src/db/pg.rs:554]: We need to process all of the tables twice - one to drop their fkey constraints , and then the tables can be dropped.

**Review by Electron100 — APPROVED** [review]

**Pull request merged**